airflow/www/views.py

```diff
@@ -1966,6 +1966,7 @@ def trigger(self, session=None):
         flash(f"Triggered {dag_id}, it should start any moment now.")
         return redirect(origin)
 
+    @provide_session
```

Comment:
Since this is an internal function, we should be able to control all cases where `session` is needed, and explicitly pass it. This decorator is thus redundant and should be removed.

Also, can we take this change to add type hints to all arguments?

Reply:
Good idea, make sense.

Done